test/fastboot-server-test.js

```diff
@@ -35,51 +56,164 @@ describe("FastBootServer", function() {
     expect(fn).to.throw(/(.+)\/fixtures\/empty-package-json\/package.json was malformed or did not contain a manifest/);
   });
 
-  it("can reload the distPath", function() {
-    var distPath = fixture('basic-app');
+  it('reads the config from package.json', function() {
+    var distPath = fixture('config-app');
 
     var server = new TestHTTPServer({
       distPath: distPath,
       port: 0
     });
 
     var promise = server.start()
-      .then(requestFirstApp)
-      .then(hotReloadApp)
-      .then(requestSecondApp)
-      .finally(cleanup);
+      .then(requestIndex(function (html) {
+        expect(html).to.match(/Config foo: bar/);
+      }))
+      .finally(cleanup(server));
+
+    var url;
+
+    return promise;
+  });
+
+  it('handles apps with config defined in app.js', function() {
+    var distPath = fixture('config-not-in-meta-app');
+
+    var server = new TestHTTPServer({
+      distPath: distPath,
+      port: 0
+    });
+
+    var promise = server.start()
+      .then(requestIndex(function(html) {
+        expect(html).to.match(/Welcome to Ember/);
+      }))
+      .finally(cleanup(server));
 
     var url;
 
     return promise;
+  });
+
+  it('prefers APP_CONFIG environment variable', function() {
+    var distPath = fixture('config-app');
+
+    var config = {
+      modulePrefix: "fastboot-test",
+      environment: "development",
+      baseURL: "/",
+      locationType: "auto",
+      EmberENV: { "FEATURES":{} },
+      APP: {
+        name: "fastboot-test",
+        version: "0.0.0+3e9fe92d",
+        autoboot: false,
+        foo: "baz"
+      },
+      exportApplicationGlobal:true
+    };
+
+    process.env.APP_CONFIG = JSON.stringify(config);
+    var server = new TestHTTPServer({
+      distPath: distPath,
+      port: 0
+    });
+
+    var promise = server.start()
+      .then(requestIndexRoute)
+      .finally(cleanup(server));
+    delete process.env.APP_CONFIG;
+
+    return promise;
 
-    function requestFirstApp(info) {
-      url = 'http://[' + info.host + ']:' + info.port + '/';
+    function requestIndexRoute(info) {
```

Comment:
Can this be removed in favor of the `requestIndex`  helper function added above?